### DISCUSSION TITLE
how to run the pipeline with different window and step size ?

### DISCUSSION THREAD

**Discussion by amitli1** [pyannote/pyannote-audio#1293] [Development]
I want to run the pipeline with a window size of 3 seconds and 1-second step.

```
from pyannote.audio import Pipeline
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization",use_auth_token="..")
diarization = pipeline("example.wav")
```

I didn't see any configuration I can pass to the pipeline.

How can I do it ?

**Comment by hbredin** [comment]
This is not supported out of the box.
You'd have to modify the code and pass your required `duration` and `step` here:

https://github.com/pyannote/pyannote-audio/blob/74939acbfa830521a434cb4068176196dd9612dc/pyannote/audio/pipelines/speaker_diarization.py#L141

**Comment by Rajsharma82** [comment]
File "C:\spreco\pyno.py", line 6, in <module>
    diarization = pipeline("eng.wav")
                  ^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not callable



 when i am trying to run this code this error is occuring
